app/assets/javascripts/Components/assignment_summary_table.jsx

```diff
@@ -29,96 +31,109 @@ class AssignmentSummaryTable extends React.Component {
         col["filterable"] = false;
         col["defaultSortDesc"] = true;
       });
+      const markingStates = getMarkingStates(res.data);
       this.setState({
         data: res.data,
         criteriaColumns: res.criteriaColumns,
         num_assigned: res.numAssigned,
         num_marked: res.numMarked,
         loading: false,
+        marking_states: markingStates,
       });
     });
   };
 
-  fixedColumns = [
-    {
-      Header: I18n.t("activerecord.models.group.one"),
-      id: "group_name",
-      accessor: "group_name",
-      Cell: row => {
-        if (row.original.result_id) {
-          const path = Routes.edit_assignment_submission_result_path(
-            this.props.assignment_id,
-            row.original.submission_id,
-            row.original.result_id
-          );
-          return <a href={path}>{row.original.group_name}</a>;
-        } else {
-          return <span>{row.original.group_name}</span>;
-        }
-      },
-      filterMethod: (filter, row) => {
-        if (filter.value) {
-          // Check group name
-          if (row._original.group_name.includes(filter.value)) {
-            return true;
+  onFilteredChange = () => {
```

Comment:
This callback function can actually take 2 arguments (https://github.com/tannerlinsley/react-table/tree/v6#fully-controlled-component). This should allow you to set `markingStateFilter` without searching through `summaryTable.state.filtered`.

Also, if the marking state column is what's changed, we don't need to recompute `markingStates`. So I think it's possible for this function to either `setState` for `marking_states` or `markingStateFilter` but not both.

Reply:
Thanks for pointing that out!

And you are correct about splitting this to only ever `setState` for one of the two options